cpp/include/cudf/strings/convert/convert_integers.hpp

```diff
@@ -78,23 +78,31 @@ std::unique_ptr<column> from_integers(
  * characters are valid for conversion to integers.
  *
  * The output row entry will be set to `true` if the corresponding string element
- * has at least one character in [-+0-9].
+ * has at least one character in [-+0-9]. The optional sign character must only be in the first
+ * position. Also, the integer component must fit within the size limits of the underlying
+ * storage type, which is provided by the int_type parameter.
  *
  * @code{.pseudo}
  * Example:
- * s = ['123', '-456', '', 'A', '+7']
- * b = s.is_integer(s)
- * b is [true, true, false, false, true]
+ * s = ['123456', '-456', '', 'A', '+7']
+ *
+ * output1 = s.is_integer(s, data_type{type_id::INT32})
+ * output1 is [true, true, false, false, true]
+ *
+ * output2 = s.is_integer(s, data_type{type_id::INT8})
+ * output2 is [false, false, false, false, true]
  * @endcode
  *
  * Any null row results in a null entry for that row in the output column.
  *
- * @param strings Strings instance for this operation.
- * @param mr Device memory resource used to allocate the returned column's device memory.
- * @return New column of boolean results for each string.
+ * @param strings  Strings instance for this operation.
+ * @param int_type Integer type used for checking underflow and overflow.
+ * @param mr       Device memory resource used to allocate the returned column's device memory.
+ * @return         New column of boolean results for each string.
  */
 std::unique_ptr<column> is_integer(
   strings_column_view const& strings,
+  data_type int_type,
```

Comment:
I think it would be better to just a 2nd `is_integer` that takes a `data_type` and leave the original one in place. This will they Python layer to decide whether to use this or not rather than force it to use this one.

Reply:
Sorry I don't get this. Do you mean put this second `is_integer` to the same place as the first (existing) `is_integer`?

Reply:
There would be two APIs like this.
```
std::unique_ptr<column> is_integer(
  strings_column_view const& strings,
  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
  
std::unique_ptr<column> is_integer(
  strings_column_view const& strings,
  data_type int_type,
  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());

```
The original code stays as it is. And you are just adding a new one.
This would all go into the same source files.

Reply:
I would need to write more tests for both APIs.

Reply:
No. You would only need to add test for the new API. The original API already has gtests.

Reply:
This `at least one` character is to match the behavior of the other `is_` APIs we have. You should not change the behavior of the original API since that is how the Python layer is coded. So no need for changing gtests for the original API.

Reply:
I see. I have modified all those original gtests for the new API, so I'll just copy it back.